### DISCUSSION TITLE
Specify provider for ENS lookup

### DISCUSSION THREAD

**Discussion by iammatthias** [wevm/wagmi#111] [Q&A]
Is it possible to specify a provider just for ENS lookup? I am building a project on Polygon, but would like to pull ENS from mainnet. 

Here is how I currently have my provider configured: 

```
  // Set up providers
  const network = {
    name: 'maticmum',
    chainId: 80001,
    _defaultProvider: providers =>
      new providers.AlchemyProvider(80001, alchemy),
  }

  const provider = () => providers.getDefaultProvider(network)
```

**Comment by gasolin** [comment]
ENS lookup seems always fetching from the mainnet

**Reply by lsbyerley** [reply]
I've noticed this also.. i have an ens domain on the rinkeby testnet and the ens domain doesn't seem to pull it in. Only when i'm on eth mainnet

**Reply by tmm** [reply]
Does your ethers.js [provider](https://wagmi-xyz.vercel.app/docs/provider#provider-optional) take into account the current network?

```tsx
import { providers } from 'ethers'
import { Provider } from 'wagmi'

// Using `chainId` allows your provider to use the current network
const provider = ({ chainId }) =>
  new providers.InfuraProvider(chainId, 'Your infura id')

const App = () => (
  <Provider provider={provider}>
    <YourRoutes />
  </Provider>
)
```

**Reply by lsbyerley** [reply]
Yeah it does, although i'm using alchemy, don't think that would matter though?

```jsx
  const provider = ({ chainId }) => {
    return new providers.AlchemyProvider(chainId, alchemyId);
  };
```

**Reply by lsbyerley** [reply]
do we need to change our alchemyId based on the chainId or can we juse the same one no matter what the chainId is?

**Reply by tmm** [reply]
No, you shouldn't need to.

Try connecting your wallet on the docs site [here](https://wagmi-xyz.vercel.app/guides/connect-wallet). If you set things up correctly, ENS lookup should respond to network changes.

https://user-images.githubusercontent.com/6759464/152718059-ccc5dbef-a31e-4f7c-a4ff-971f2b1fadfd.mp4
